ksql-engine/src/test/java/io/confluent/ksql/structured/SchemaKStreamTest.java

```diff
@@ -160,4 +167,45 @@ public void testSelectKey() throws Exception {
 
   }
 
+  @Test
+  public void testGroupByKey() {
+    String selectQuery = "SELECT col0, col1 FROM test1 WHERE col0 > 100;";
+    PlanNode logicalPlan = planBuilder.buildLogicalPlan(selectQuery);
+    initialSchemaKStream = new SchemaKStream(logicalPlan.getTheSourceNode().getSchema(), kStream,
+        ksqlStream.getKeyField(), new ArrayList<>(),
```

Comment:
Would it make sense to assert in this test that  `ksqlStream.keyField()` is actually `col0`? This is the crux of the patch if I understand correctly.